plugins/dummy_rand_data_sdk/src/main.rs

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use hipcheck_sdk::{
+	async_trait, error::Error, from_str, plugin_engine::PluginEngine, plugin_server::PluginServer,
+	proto::GetQuerySchemasResponse, JsonSchema, NamedQuery, Plugin, Query, Value,
+};
+
+static GET_RAND_KEY_SCHEMA: &str = include_str!("../schema/query_schema_get_rand.json");
+static GET_RAND_OUTPUT_SCHEMA: &str = include_str!("../schema/query_schema_get_rand.json");
+
+fn reduce(input: u64) -> u64 {
+	input % 7
+}
+
+/// Plugin that queries hipcheck takes a `Value::Number` as input and performs following steps:
+/// - ensures input is u64
+/// - % 7 of input
+/// - queries `hipcheck` for sha256 of (% 7 of input)
+/// - returns `Value::Number`, where Number is the first `u8` in the sha256
+///
+/// Goals of this plugin
+/// - Verify `salsa` memoization is working (there should only ever be 7 queries made to `hipcheck`)
+/// - Verify plugins are able to query `hipcheck` for additional information
+#[derive(Clone, Debug)]
+struct RandDataPlugin {
+	schema: GetQuerySchemasResponse,
+}
+
+#[async_trait]
+impl Query for RandDataPlugin {
+	fn input_schema(&self) -> JsonSchema {
+		from_str(self.schema.key_schema.as_str()).unwrap()
+	}
+
+	fn output_schema(&self) -> JsonSchema {
+		from_str(self.schema.output_schema.as_str()).unwrap()
+	}
+
+	async fn run(
+		&self,
+		engine: &mut PluginEngine,
+		input: Value,
+	) -> hipcheck_sdk::error::Result<Value> {
+		let Value::Number(num_size) = input else {
+			return Err(Error::UnexpectedPluginResponseDataFormat);
```

Comment:
the error type saying `Response` when this is checking the type of the "request" key feels wrong